src/MediaPresenter.ts

```diff
@@ -0,0 +1,153 @@
+import * as mobx from "mobx";
+
+const MAXLEN = 10;
+
+export class MediaPresenter {
+  constructor() {
+    mobx.makeObservable(this);
+  }
+  // make need to change to .deep
+  @mobx.observable.deep
+  files: File[] = [];
+
+  @mobx.observable.deep
+  durations: number[] = [];
+  // for now its a 1D array but can be expanded as [][] to play different parts
+  // this is for videos, which parts of it have been played
+  played: number[] = [];
+
+  customOrder: boolean = false;
+
+  static audioSound: number = 3;
+
+  static isImage(file: File): boolean {
+    const imageFormat = new RegExp("image/*");
+    return imageFormat.test(file.type);
+  }
+
+  static isVideo(file: File): boolean {
+    const videoFormat = new RegExp("video/*");
+    return videoFormat.test(file.type);
+  }
+
+  @mobx.action
+  addFile(newFile: File): boolean {
+    if (MediaPresenter.isImage(newFile) || MediaPresenter.isVideo(newFile)) {
+      this.files.push(newFile);
+      this.durations.push(0);
+      this.played.push(0);
+      return true;
+    }
+    return false;
+  }
+
+  @mobx.action
+  removeFile(fileIndex: number): void {
+    if (this.files.length === 0) {
+      return;
+    }
+    let index = fileIndex % this.filesLength;
+    if (index > -1) {
+      this.files.splice(index, 1);
+      this.durations.splice(index, 1);
+      this.played.splice(index, 1);
+    }
+  }
+
+  setCustomOrder(value: boolean) {
+    this.customOrder = value;
+  }
+
+  @mobx.action
+  setDuration(fileIndex: number, duration: number): void {
+    this.durations[fileIndex % this.filesLength] = duration;
+  }
+
+  getDuration(fileIndex: number): number {
+    return this.durations[fileIndex % this.filesLength];
+  }
+
+  getFileName(fileIndex: number): string {
+    return this.files[fileIndex % this.filesLength].name;
+  }
+
+  getFile(fileIndex: number): File {
+    return this.files[fileIndex % this.filesLength];
+  }
+
+  @mobx.computed
+  get filesLength(): number {
+    return this.files.length;
+  }
+
+  @mobx.action
+  incrementFilePlayed(fileIndex: number, seconds: number): void {
+    let index = fileIndex % this.filesLength;
+    console.log("in increment " + this.played[index]);
+    console.log(this.durations[index]);
+    this.played[index] = (this.played[index] + seconds) % this.durations[index];
+  }
+
+  @mobx.action
+  resetFilePlayed(): void {
```

Comment:
as this is resetting all files, the name could be better described as `resetAllPlayedFiles` or just `resetFilesPlayed` (the use of the singular implies that it only affects one file)